src/DependencyInjection/Compiler/BlockHelperCompilerPass.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @internal
+ *
+ * NEXT_MAJOR: remove this class
+ */
+final class BlockHelperCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        $cacheBlocks = $container->getParameter('sonata_block.cache_blocks');
+        if (0 !== \count($cacheBlocks)) {
+            @trigger_error(
+                'Defining cache blocks is deprecated since sonata-project/block-bundle 4.x and will not be supported anymore in 5.0.',
+                \E_USER_DEPRECATED
+            );
+
+            return;
+        }
+
+        $blockHelperDefinition = $container->getDefinition('sonata.block.templating.helper');
+        $blockHelperDefinition->setArguments([
```

Comment:
This is a nice idea.
I just feel like we should do the opposite.

 - Update our 'default' config to pass the service needed by the new signature
 - In this compiler pass, if there is no cache blocks, return.
 - In this compiler pass, if there are cache blocks, trigger a deprecation and update the definition.

This way, in the next major we just have to remove this (instead having to update the service config too).

Reply:
Indeed that's better 👍

Reply:
changed it